### PR TITLE
Node based pre/post install scripts I462

### PR DESCRIPTION
**Pull request by heyMP**

### CODE REVIEW

File: packages/generator-wcfactory/generators/factory/templates/scripts/postinstall.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+const path = require('path')
+const pkgJson = require('../package.json')
+
+// get the factory npm org name as it is stored in node_modules
+const factoryName = pkgJson.wcfactory.orgNpm
+/**
+ * @todo this should be dynaimc probably
+ */
+// get elements directory in the
+const elementsDir = path.join(__dirname, '../elements')
```
Comment (ChristianMurphy) [inline packages/generator-wcfactory/generators/factory/templates/scripts/postinstall.js]: Would it be possible to use `lerna list --json` to get the active component folders?

File: packages/generator-wcfactory/generators/factory/templates/scripts/postinstall.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  // move on to the next thing if there is an error
+  if (err) return
+  folders.forEach(element => {
+    const elementDir = path.join(elementsDir, element)
+
+      // re-symlink the node_modules dir
+      // make sure that this is a directory
+      const nodeModulesDir = path.join(elementDir, 'node_modules')
+      // check if this element has a node modules dir in it
+      if (fs.existsSync(nodeModulesDir)) {
```
Comment (ChristianMurphy) [inline packages/generator-wcfactory/generators/factory/templates/scripts/postinstall.js]: `fs.exists` has been deprecated, `fs.access` or `fs.stat` would be a safer alternative

File: packages/generator-wcfactory/generators/factory/templates/scripts/postinstall.js (hunk 1 above)
Comment (heyMP) [reply packages/generator-wcfactory/generators/factory/templates/scripts/postinstall.js]: I'm thinking that we might want to specify it in some other way.  We might have folders that don't contain web components in the same monorepo.  For instance, we have services that we don't want these scripts to affect.

Maybe a setting within package.json under the `wcfactory` property?

Example:
```
{
  "name": "<%= orgNpm %>/<%= name %>",
  "version": "0.0.0",
  "description": "<%= description %>",
  "wcfactory": {
    "installCleanupDirectories": [
      "elements"
    ]
  },
```

File: packages/generator-wcfactory/generators/factory/templates/scripts/postinstall.js (hunk 2 above)
Comment (heyMP) [reply packages/generator-wcfactory/generators/factory/templates/scripts/postinstall.js]: I don't think those are in fs-extra yet?

File: packages/generator-wcfactory/generators/factory/templates/scripts/postinstall.js (hunk 1 above)
Comment (btopro) [reply packages/generator-wcfactory/generators/factory/templates/scripts/postinstall.js]: even if you don't have components in here though the point was to run the yarn scripts via a ui utility right? So as long as whatever it is has package.json then it should be fair game?

File: packages/generator-wcfactory/generators/factory/templates/scripts/postinstall.js (hunk 2 above)
Comment (ChristianMurphy) [reply packages/generator-wcfactory/generators/factory/templates/scripts/postinstall.js]: https://github.com/jprichardson/node-fs-extra/blob/8e0879110fbdd597c36602fe3b81ef03a4b3ec7a/lib/fs/index.js#L8

**Comment by btopro** [comment]
has this been tested?

**Comment by heyMP** [comment]
I haven't tested it in a few months but it was working.  We still need to implement christian's changes but those can probably be converted to open issues.

**Pull request closed without merging**